flocker/control/test/test_protocol.py

```diff
@@ -451,3 +518,295 @@ class FakeAgentInterfaceTests(iconvergence_agent_tests_factory(
     """
     ``IConvergenceAgent`` tests for ``FakeAgent``.
     """
+
+SEND_REQUEST = ActionType(
+    u'test:send_request',
+    [],
+    [],
+    u'client makes request to server.'
+)
+
+HANDLE_REQUEST = ActionType(
+    u'test:handle_request',
+    [],
+    [],
+    u'server receives request from client.'
+)
+
+
+class ClientProcess(object):
+    logger = None
+
+    def send_request(self, server, **kwargs):
+        """
+        Send request.
+        """
+        with SEND_REQUEST(self.logger) as action:
+            return server.handle_request(
+                eliot_context=action.serialize_task_id(),
+                **kwargs
+            )
+
+
+class ServerProcess(object):
+    logger = None
+
+    @with_eliot_context
```

Comment:
The solution is to move `server.handle_request` out of the `with SEND_REQUEST` block:

```
with SEND_REQUEST(self.logger) as action:
    task_id = action.serialize_task_id()
return server.handle_request(eliot_context=task_id, **kwargs)
```